statsd/buffer_test.go

```
@@ -0,0 +1,272 @@
+	buffer.elementCount = 1
+	pos, err = buffer.writeAggregated([]byte("h"), "namespace.", []string{"tag:tag"}, "metric", []float64{1, 2, 3, 4}, "", 12, -1, 1)
+	assert.Equal(t, errBufferFull, err)
+
+	// not enough size to start serializing (tags and header too big)
+	buffer = newStatsdBuffer(4, 1)
+	pos, err = buffer.writeAggregated([]byte("h"), "namespace.", []string{"tag:tag"}, "metric", []float64{1, 2, 3, 4}, "", 12, -1, 1)
+	assert.Equal(t, errBufferFull, err)
+
+	// not enough size to serializing one message
+	buffer = newStatsdBuffer(29, 1)
+	pos, err = buffer.writeAggregated([]byte("h"), "namespace.", []string{"tag:tag"}, "metric", []float64{1, 2, 3, 4}, "", 12, -1, 1)
```

golangci-lint (1.62.2)

ineffectual assignment to pos

(ineffassign)